zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import random
+
+from six.moves.urllib import error
+from six.moves.urllib.request import urlopen
+
+from zulip_bots.test_lib import BotTestCase
+
+class TestGoogleSearchBot(BotTestCase):
+    bot_name = 'googlesearch'
+
+    # Simple query
+    def test_bot(self):
+        self.assert_bot_response({'content':'zulip'},{'content':'Success: https://zulipchat.com/'}, 'send_reply')
+
+    # Help without typing anything
+    def test_bot_help_none(self):
+        help_message = "To use this bot, start messages with @mentioned-bot, \
+                    followed by what you want to search for. If \
+                    found, Zulip will return the first search result \
+                    on Google.\
+                    \
+                    An example message that could be sent is:\
+                    '@mentioned-bot zulip' or \
+                    '@mentioned-bot how to create a chatbot'."
+        self.assert_bot_response({'content':''},{'content':help_message}, 'send_reply')
+
+    # Help from typing 'help'
+    def test_bot_help(self):
+        help_message = "To use this bot, start messages with @mentioned-bot, \
+                    followed by what you want to search for. If \
+                    found, Zulip will return the first search result \
+                    on Google.\
+                    \
+                    An example message that could be sent is:\
+                    '@mentioned-bot zulip' or \
+                    '@mentioned-bot how to create a chatbot'."
+        self.assert_bot_response({'content':'help'},{'content':help_message}, 'send_reply')
+
+    # A google search with no results
+    def test_bot_no_results(self):
+        ranstr = ''
+        characters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890'
+        for i in range(0, 100):
+            ranstr += characters[random.randrange(len(characters))]
+        self.assert_bot_response({'content':ranstr},{'content':'Error: Search failed.'}, 'send_reply')
+
+    # Tests to see if it can connect to google
+    def test_connection(self):
+        try:
+            urlopen('http://216.58.192.142', timeout=1)
```

Comment:
Hi, in our tests we want to avoid making real connections to the internet.  We want to use mock_http_conversation instead.  Examples can be found in the giphy tests.

Reply:
Ah, ok

Reply:
Well the point of that test was to test connection, so I'll just get rid of it.

Reply:
Would I have to also do this for `test_bot_no_results` aswell?